Describe adding insert support in a connector (#13046)
trinodb/trino: docs/src/main/sphinx/develop/insert.rst

=============================================
Supporting ``INSERT`` and ``CREATE TABLE AS``
=============================================

To support ``INSERT``, a connector must implement:

> * ``beginInsert()`` and ``finishInsert`` from the ``ConnectorMetadata``

Reviewer: `finishInsert` -> `finishInsert()`

  interface;
* a ``ConnectorPageSinkProvider`` that receives a table handle and returns
  a ``ConnectorPageSink``.

When executing an ``INSERT`` statement, the engine calls the ``beginInsert()``

> method in the connector, which receives a table handle and a list of column.

Reviewer: `column` -> `columns`

> It should return a ``ConnectorInsertTableHandle``, that can carry any

Reviewer 2: The grammar of this sentence is off. cc @mosabua

> connector specific information and will be passed to the page sink provider.

Reviewer 3: ```suggestion
connector specific information, and it is passed to the page sink provider.
```

Author: I forgot about using the present tense, but I wanted to ask about passive voice - is `it is passed` passive and should we change it to say that the engine passes it?

> The page sink provider creates a page sink, that should accept ``Page`` objects.

Reviewer: _page sink provider_  --> `PageSinkProvider`

> The ``PageSinkProvider`` creates a page sink, that should accept ``Page`` objects.

Reviewer 3: ```suggestion
The ``PageSinkProvider`` creates a page sink, that accepts ``Page`` objects.
```

> When all the pages for a specific split have been processed, the Trino engine

Reviewer 3: ```suggestion
When all the pages for a specific split have been processed, Trino calls
```

calls ``ConnectorPageSink.finish()``, which returns a ``Collection<Slice>``

> of "fragments" representing connector-specific information about

Reviewer 3: drop the ""

Author: I copied this from the existing `delete and update` section. I fixed it there too, in a separate commit.

the processed rows.

> When all pages for all splits have been processed, the Trino engine calls

Reviewer 3: Just use "Trino" in general .. dont talk about "the Trino engine"

``ConnectorMetadata.finishInsert()``, passing a collection containing all
the "fragments" from all the splits.  The connector does what is required
to finalize the operation, for example, committing the transaction.

> To support ``CREATE TABLE AS``, the ``ConnectorPageSinkProvider`` should

Reviewer 3: must also
